lib/ruby_smb/client/negotiation.rb

```diff
@@ -9,8 +9,8 @@ module Negotiation
       # It also keeps track of the negotiated dialect.
       #
       # @return [void]
-      def negotiate
-        request_packet  = negotiate_request
+      def negotiate(negotiate_packet: nil)
```

Comment:
I’m not sure about adding `negotiate_packet` argument to `#negotiate`. It might be confusing since it is only used for a very limited use case. Did you consider moving this logic directly to the module you’re working on?
For example, I think it is the only part you’ll need:
```ruby
raw_response    = client.send_recv(custom_negotiate_packet)
response_packet = client.negotiate_response(raw_response)
response_packet.dialects = request_packet.dialects
# parse_negotiate_response used section:
client.smb1_response(response_packet)
challenge = response_packet.data_block.challenge
```

Reply:
Yeah this is a nicer way of doing it will change over 👍